crates/ruff_linter/src/rules/pycodestyle/rules/compound_statements.rs

```diff
@@ -294,40 +285,40 @@ pub(crate) fn compound_statements(
                 with = None;
             }
             TokenKind::Case => {
-                case = Some((range.start(), range.end()));
+                case = Some((token.start(), token.end()));
```

Comment:
This is unrelated to this PR but I wonder if we can use a single variable to track all clauses with a body (`case`, `class`, `elif`, ...) since we always test them at once and reset them at once.